components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/deployers/config/ServiceProviderConfigDeployer.java

```diff
@@ -55,37 +55,51 @@ public ServiceProviderConfigDeployer(int tenantId, String id) {
         this.id = id;
     }
 
+    /**
+     * Used to retrieve Id.
+     *
+     * @return id
+     */
     public String getId() {
+
         return id;
     }
 
     /**
      * Deploy the configuration.
      *
-     * @param configurationFileStream
-     * @throws RemoteFetchCoreException
+     * @param configurationFileStream ConfigurationFileStream of cloned application
+     * @throws RemoteFetchCoreException RemoteFetchCoreException
+     * @throws IOException              IOException
      */
     @Override
     public void deploy(ConfigurationFileStream configurationFileStream) throws RemoteFetchCoreException, IOException {
 
         ServiceProvider previousServiceProvider = null;
-
-        Tenant tenant = this.getTenantDomain();
-        String userName = tenant.getAdminName();
-        startTenantFlow(tenant.getDomain(), userName);
+        String userName = this.getTenantAdminName();
+        String tenantDomain = this.getTenantDomain();
+        if (log.isDebugEnabled()) {
+            log.debug("Tenant admin name " + userName + " retrieved for tenant domain " + tenantDomain);
+        }
+        startTenantFlow(tenantDomain, userName);
```

Comment:
Ideally we should improve this to use the user who add the configuration instead of using tenant admin.

Reply:
Agreed. I created an issue to track it as improvement. https://github.com/wso2-extensions/identity-fetch-remote/issues/39